### PR TITLE
MAT-6390: Implement valuesets page. correct select to autocomplete, d…

### PR DESCRIPTION
**Pull request by mcmcphillips**
Implement valuesets page. correct select to autocomplete, disable tab for valuesets if not owner

## MADiE PR

Jira Ticket: [MAT-6390](https://jira.cms.gov/browse/MAT-6390)
(Optional) Related Tickets:
Jira Ticket: [MAT-6389](https://jira.cms.gov/browse/MAT-6389)
Jira Ticket: [MAT-6917](https://jira.cms.gov/browse/MAT-6917)

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/terminologySectionPanel/ValueSets/Search/MultipleSelectDropDown.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                checkedIcon={checkedIcon}
+                style={{ marginRight: 8 }}
+                checked={selected}
+              />
+              {option.label}
+            </li>
+          );
+        }}
+        renderInput={(params) => {
+          const { inputProps } = params;
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/ValueSets/Search/MultipleSelectDropDown.tsx]: looks like an unused variable, Not sure if the inputProps require a data-testId to the TextField. Our Automation scripts seems to be using them a lot.

File: src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import MultipleSelectDropDown from "./MultipleSelectDropDown";
+
+const SelectSearchCategories = ({
+  formikFieldProps,
+  handleChange,
+  options,
+}) => {
+  return (
+    <div>
+      <MultipleSelectDropDown
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx]: Are we expecting to have more child Components in SelectSearchCategories ?

File: src/terminologySectionPanel/ValueSets/ValueSets.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        <div className="search-container">
+          {groupedRows.map((group, index) => (
+            // each row
+            <div key={index} className="search-row">
+              {/* Each row touple, what about odd cases? We need logic to conditionally Add an empty div for formatting */}
+              {group.map(({ value }, index, a) => {
+                if (a.length === 1) {
+                  return (
+                    <>
+                      <SearchField
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/ValueSets/ValueSets.tsx:118]: If possible having screenshots/screen recordings attached to the PR will help in visualizing the functionality while reviewing.

File: src/terminologySectionPanel/TerminologySectionPanelNavTabs.tsx [hunk 4]
```diff
@@ -19,14 +19,15 @@ export default function TerminologySectionPanelNavTabs(props: NavTabProps) {
...
-      {QDMValueSetSearch && canEdit && (
+      {QDMValueSetSearch && (
         <Tab
           tabIndex={0}
           aria-label="Value Sets"
           type="B"
           label="Value Sets"
           data-testid="valueSets-tab"
           value="valueSets"
+          disabled={!canEdit}
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/TerminologySectionPanelNavTabs.tsx:30]: I wonder if we should even disabled this tab for non-owners, Even a non-owner or shared user should still be able to see the applied valueSets in a CQL

File: src/terminologySectionPanel/ValueSets/ValueSets.tsx (hunk 3 above)
Comment (RohitKandimalla) [reply src/terminologySectionPanel/ValueSets/ValueSets.tsx:118]: IDK if this is my local issue and I might not have everything up-to-date but looks like action buttons are mis-aligned. 
image

File: src/terminologySectionPanel/ValueSets/ValueSets.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        formik.setFieldValue(value, "");
+      }
+    });
+  };
+  return (
+    <form
+      id="madie-editor-value-sets"
+      data-testid="madie-editor-value-sets"
+      onSubmit={(e) => {
+        e.preventDefault();
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/ValueSets/ValueSets.tsx]: onSubmit={formik.handleSubmit} Formik should be handling the default actions

File: src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        formControl={formikFieldProps}
+        {...formikFieldProps}
+        id="search-by-category"
+        label="Search By Category"
+        required={false}
+        disabled={false}
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx]: It is better if we can disable fields using canEdit

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx [hunk 7]
```diff
@@ -120,26 +127,31 @@ export default function CodeSection({
...
-                    }}
-                    label="Code System"
-                    id={"code-system-selector"}
-                    inputProps={{
-                      "data-testid": "code-system-selector-input",
-                    }}
-                    data-testid={"code-system-selector"}
-                    SelectDisplayProps={{
-                      "aria-required": "true",
+                  <MultipleSelectDropDown
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx]: Why does Code System Selection have to be multiple ? Is this expected ? but I don't think we can search for a code in multiple code systems

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/SelectCodeSystem.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import MultipleSelectDropDown from "../../../ValueSets/Search/MultipleSelectDropDown";
+
+const SelectCodeSystem = ({ formikFieldProps, handleChange, options }) => {
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/SelectCodeSystem.tsx]: Can't seem to find an usage for this SelectCodeSystem component, I think you directly added multiSelectDropDown for CodeSystem selection

File: src/terminologySectionPanel/ValueSets/ValueSets.tsx (hunk 3 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/ValueSets/ValueSets.tsx:112]: image


I wonder what can cause this difference

File: src/terminologySectionPanel/TerminologySectionPanelNavTabs.tsx (hunk 4 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/TerminologySectionPanelNavTabs.tsx:30]: > Given: A QDM measure, I am on the CQL Editor screen and feature flag called "QDMValueSetSearch" is set to "true" and I am NOT an owner and measure is NOT shared with me
When: I look to the right of the CQL Editor
Then: I see a Value Sets tab but it is disabled (all the contents of the tab are disabled)

[MAT-6389](https://jira.cms.gov/browse/MAT-6389)

File: src/terminologySectionPanel/ValueSets/ValueSets.tsx (hunk 5 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/ValueSets/ValueSets.tsx]: modified to empty handleSubmit

File: src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx]: no. selectSearchCategories deleted. using reference one level up.

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx (hunk 7 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx]: It is not multiple. The multiple prop being passed the that component is false. 

Naming of component has been to ControlledAutoComplete for clarity

File: src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx (hunk 6 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/ValueSets/Search/SelectSearchCategories.tsx]: This page is behind a can edit one level up. 

TerminologySectionPanelNavTabs has the tab disabled if user cannot edit. 
It should not be possible to reach this code without canEdit

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/SelectCodeSystem.tsx (hunk 8 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/SelectCodeSystem.tsx]: Removed unused component

File: src/terminologySectionPanel/ValueSets/Search/MultipleSelectDropDown.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/terminologySectionPanel/ValueSets/Search/MultipleSelectDropDown.tsx]: Data-testid is already programmatically added to the TextField component as "id-input" 
We previously had to pass things manually before the inclusion of this code


https://github.com/MeasureAuthoringTool/madie-design-system/blob/main/react/components/TextField/index.jsx#L23-L40

**Review by RohitKandimalla — APPROVED** [review]
